### PR TITLE
[Api][Homepage] menu taxons

### PR DESCRIPTION
**Pull request by AdamKasp**
| Q               | A
| --------------- | -----
| Branch?         | master
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | part of #11250 
| License         | MIT

### CODE REVIEW

File: src/Sylius/Bundle/ApiBundle/spec/DataProvider/TaxonFromChannelMenuTaxonCollectionDataProviderSpec.php [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+    function it_throws_an_exception_if_context_has_not_channel(): void
+    {
+        $this
+            ->shouldThrow(\InvalidArgumentException::class)
+            ->during('getCollection', [TaxonInterface::class, 'get_from_menu_taxon', []])
+        ;
+    }
+
+    function it_provides_taxon_from_channel_menu_taxon(
```
Comment (GSadee) [inline src/Sylius/Bundle/ApiBundle/spec/DataProvider/TaxonFromChannelMenuTaxonCollectionDataProviderSpec.php:42]:
```suggestion
    function it_provides_taxons_from_channel_menu_taxon(
```

File: src/Sylius/Bundle/ApiBundle/spec/SerializerContextBuilder/LocaleContextBuilderSpec.php [hunk 2]
```diff
@@ -25,7 +25,7 @@ function let(
        $this->beConstructedWith($decoratedSerializerContextBuilder, $localeContext);
    }
 
-   function it_updates_context_when_locale_context_has_locale(
+   function it_updates_an_context_when_locale_context_has_locale(
```
Comment (GSadee) [inline src/Sylius/Bundle/ApiBundle/spec/SerializerContextBuilder/LocaleContextBuilderSpec.php:28]:
```suggestion
   function it_updates_a_context_when_locale_context_has_locale(
```

File: src/Sylius/Bundle/ApiBundle/spec/DataProvider/TaxonFromChannelMenuTaxonCollectionDataProviderSpec.php [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        $this->beConstructedWith($taxonRepository);
+    }
+
+    function it_supports_only_get_from_menu_taxon_operation_on_taxon_entity(): void
+    {
+        $this->supports(TaxonInterface::class, 'get_from_menu_taxon')->shouldReturn(true);
+        $this->supports(ProductInterface::class, 'get_from_menu_taxon')->shouldReturn(false);
+    }
+
+    function it_throws_an_exception_if_context_has_not_channel(): void
```
Comment (GSadee) [inline src/Sylius/Bundle/ApiBundle/spec/DataProvider/TaxonFromChannelMenuTaxonCollectionDataProviderSpec.php:34]:
```suggestion
    function it_throws_an_exception_if_context_has_no_channel(): void
```

**Comment by GSadee** [comment]
Thanks, Adam! :1st_place_medal:

File: src/Sylius/Bundle/ApiBundle/Resources/config/api_resources/Taxon.xml [hunk 4]
```diff
@@ -15,17 +14,33 @@
...
             <collectionOperation name="post">
+                <attribute name="path">admin/taxons</attribute>
                 <attribute name="denormalization_context">
                     <attribute name="groups">taxon:create</attribute>
                 </attribute>
             </collectionOperation>
+
+            <collectionOperation name="get_from_menu_taxon">
+                <attribute name="method">GET</attribute>
+                <attribute name="path">/shop/taxons/from-menu-taxon</attribute>
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ApiBundle/Resources/config/api_resources/Taxon.xml:29]: Why didn't we used `shop/taxons/` as a path?

```suggestion
                <attribute name="path">shop/taxons/from-menu-taxon/</attribute>
```

**Pull request merged**